pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -213,17 +210,22 @@ type Controller struct {
 
 // NewController creates a new Kubernetes controller
 // Created by bootstrap and multicluster (see secretcontroler).
-func NewController(client kubernetes.Interface, metadataClient metadata.Interface, options Options) *Controller {
-	log.Infof("Service controller watching namespace %q for services, endpoints, nodes and pods, refresh %s",
-		options.WatchedNamespaces, options.ResyncPeriod)
-
-	watchedNamespaceList := strings.Split(options.WatchedNamespaces, ",")
+func NewController(
+	client kubernetes.Interface,
```

Comment:
We could in theory pass kube.Clients here, but passing the full set of dependencies has precedent:
https://github.com/kubernetes/kubernetes/blob/c6fd25d10032b22ca28e437957390f89a4476ecf/pkg/controller/volume/attachdetach/attach_detach_controller.go#L105